### ISSUE TITLE
setup.py sets upper bound on `importlib-metadata`

### ISSUE THREAD

**Issue by dror-weiss** [aws/sagemaker-python-sdk#3147]
**Describe the bug**
The `importlib-metadata<2.0` requirement upper bound in `setup.py` leads to dependency deadlock

**To reproduce**
Require to install `streamlit==1.9.2`,  `pytorch-lightning==1.6.3` and `sagemaker==2.92.1`

**Expected behavior**
`setup.py` without upper bound

**Screenshots or logs**
```bash
Could not find a version that matches importlib-metadata<2.0,>=1.4,>=4.4 (from sagemaker==2.92.1->face-animation==1.0->-r requirements/dev.in (line 3))
Tried: 0.0, 0.1, 0.1, 0.2, 0.2, 0.3, 0.3, 0.4, 0.4, 0.5, 0.5, 0.6, 0.6, 0.7, 0.7, 0.8, 0.8, 0.9, 0.9, 0.10, 0.10, 0.11, 0.11, 0.12, 0.12, 0.13, 0.13, 0.14, 0.14, 0.15, 0.15, 0.16, 0.16, 0.17, 0.17, 0.18, 0.18, 0.19, 0.19, 0.20, 0.20, 0.21, 0.21, 0.22, 0.22, 0.23, 0.23, 1.0.0, 1.0.0, 1.1.0, 1.1.0, 1.1.1, 1.1.1, 1.1.2, 1.1.2, 1.1.3, 1.1.3, 1.2.0, 1.2.0, 1.3.0, 1.3.0, 1.4.0, 1.4.0, 1.5.0, 1.5.0, 1.5.1, 1.5.1, 1.5.2, 1.5.2, 1.6.0, 1.6.0, 1.6.1, 1.6.1, 1.7.0, 1.7.0, 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.1.1, 2.1.1, 2.1.2, 2.1.2, 2.1.3, 2.1.3, 3.0.0, 3.0.0, 3.1.0, 3.1.0, 3.1.1, 3.1.1, 3.2.0, 3.2.0, 3.3.0, 3.3.0, 3.4.0, 3.4.0, 3.5.0, 3.5.0, 3.6.0, 3.6.0, 3.7.0, 3.7.0, 3.7.1, 3.7.1, 3.7.2, 3.7.2, 3.7.3, 3.7.3, 3.8.0, 3.8.0, 3.8.1, 3.8.1, 3.8.2, 3.8.2, 3.9.0, 3.9.0, 3.9.1, 3.9.1, 3.10.0, 3.10.0, 3.10.1, 3.10.1, 4.0.0, 4.0.0, 4.0.1, 4.0.1, 4.1.0, 4.1.0, 4.2.0, 4.2.0, 4.3.0, 4.3.0, 4.3.1, 4.3.1, 4.4.0, 4.4.0, 4.5.0, 4.5.0, 4.6.0, 4.6.0, 4.6.1, 4.6.1, 4.6.2, 4.6.2, 4.6.3, 4.6.3, 4.6.4, 4.6.4, 4.7.0, 4.7.0, 4.7.1, 4.7.1, 4.8.0, 4.8.0, 4.8.1, 4.8.1, 4.8.2, 4.8.2, 4.8.3, 4.8.3, 4.9.0, 4.9.0, 4.10.0, 4.10.0, 4.10.1, 4.10.1, 4.11.0, 4.11.0, 4.11.1, 4.11.1, 4.11.2, 4.11.2, 4.11.3, 4.11.3, 4.11.4, 4.11.4
There are incompatible versions in the resolved dependencies:
  importlib-metadata>=1.4 (from streamlit==1.9.2->streamlit-cropper==0.1.3->private_package==1.0->-r requirements/dev.in (line 3))
  importlib-metadata>=4.4 (from markdown==3.3.7->tensorboard==2.9.0->pytorch-lightning==1.6.3->private_package==1.0->-r requirements/dev.in (line 3))
  importlib-metadata<2.0,>=1.4.0 (from sagemaker==2.92.1->private_package==1.0->-r requirements/dev.in (line 3))
```

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 2.92.1

**Comment by jmahlik** [comment]
Importlib-metadata is currently at release 4.11.4 on pypi. So that should be changed for sure.

I also wanted to note, there appears to have been upper bounds set on boto3, numpy, and protobuf3-to-dict in the most recent release as well. These are likely to cause conflicts in the future. Might want to look in to removing the upper bounds on some of these.

https://github.com/aws/sagemaker-python-sdk/compare/v2.92.0...v2.92.1

```
required_packages = [
    "attrs==20.3.0",
    "boto3>=1.20.21,<2.0",
    "google-pasta",
    "numpy>=1.9.0,<2.0",
    "protobuf>=3.1,<4.0",
    "protobuf3-to-dict>=0.1.5,<1.0",
    "smdebug_rulesconfig==1.0.1",
    "importlib-metadata>=1.4.0,<2.0",
    "packaging>=20.0",
    "pandas",
    "pathos",
```

**Comment by trungleduc** [comment]
Fixed in #3985
